pkg/asset/cluster/openstack/openstack.go

```diff
@@ -48,3 +103,87 @@ func Metadata(infraID string, config *types.InstallConfig) *openstack.Metadata {
 		},
 	}
 }
+
+func replaceBootstrapIgnitionInTFVars(ctx context.Context, tfvarsFile *asset.File, installConfig *installconfig.InstallConfig, clusterID *installconfig.ClusterID) error {
+	if tfvarsFile == nil {
+		return fmt.Errorf("missing tfvars file")
+	}
+
+	var tfvars map[string]json.RawMessage
+	if err := json.Unmarshal(tfvarsFile.Data, &tfvars); err != nil {
+		return fmt.Errorf("unable to decode tfvars: %w", err)
+	}
+
+	bootstrapIgnitionJSON, ok := tfvars["openstack_bootstrap_shim_ignition"]
+	if !ok {
+		return fmt.Errorf("bootstrap's Ignition file not found")
+	}
+
+	var bootstrapIgnition string
+	if err := json.Unmarshal(bootstrapIgnitionJSON, &bootstrapIgnition); err != nil {
+		return fmt.Errorf("failed to decode bootstrap's Ignition")
+	}
+
+	bootstrapShim, err := buildBootstrapShim(ctx, installConfig.Config.Platform.OpenStack.Cloud, clusterID.InfraID, installConfig.Config.Proxy, []byte(bootstrapIgnition))
+	if err != nil {
+		return fmt.Errorf("failed to build bootstrap's Ignition shim: %w", err)
+	}
+
+	bootstrapShimJSON, err := json.Marshal(bootstrapShim)
+	if err != nil {
+		return fmt.Errorf("failed to encode bootstrap's Ignition shim: %w", err)
+	}
+
+	tfvars["openstack_bootstrap_shim_ignition"] = bootstrapShimJSON
+
+	b, err := json.MarshalIndent(tfvars, "", "  ")
+	if err != nil {
+		return fmt.Errorf("unable to encode tfvars: %w", err)
+	}
+	tfvarsFile.Data = b
+	return nil
+}
+
+func replaceBootstrapIgnitionInManifests(ctx context.Context, manifests *[]client.Object, installConfig *installconfig.InstallConfig, clusterID *installconfig.ClusterID) error {
+	const (
+		boostrapIgnitionSecretValueKey  = "value"
+		boostrapIgnitionSecretFormatKey = "format"
+	)
+
+	var (
+		bootstrapIgnitionSecret      *corev1.Secret
+		bootstrapIgnitionSecretIndex int
+	)
+
+	for i := range *manifests {
+		if (*manifests)[i].GetName() == clusterID.InfraID+"-bootstrap" {
+			var ok bool
+			bootstrapIgnitionSecret, ok = (*manifests)[i].(*corev1.Secret)
+			if ok {
+				bootstrapIgnitionSecretIndex = i
+				break
+			}
+		}
+	}
+	if bootstrapIgnitionSecret == nil {
+		return fmt.Errorf("bootstrap Ignition not found")
+	}
+
+	if format := bootstrapIgnitionSecret.Data[boostrapIgnitionSecretFormatKey]; bytes.Equal(format, []byte("ignition")) {
+		return fmt.Errorf("bootstrap Ignition format not recognised: %q", format)
```

Comment:
I'm seeing the following error:
`FATAL failed to fetch Cluster: failed to generate asset "Cluster": bootstrap Ignition format not recognised: "ignition"`

We probably need to decode the format before checking if the content is `ignition`

Reply:
hmm it looks like this should be `!bytes.Equal`, with a negation